### DISCUSSION TITLE
provider_group_id is unique at what level/namespace?

### DISCUSSION THREAD

**Discussion by theogaraj** [CMSgov/price-transparency-guide#729] [Q&A] [answered]
[This portion of the schema](https://github.com/CMSgov/price-transparency-guide/tree/master/schemas/in-network-rates#provider-reference-object) defines the Provider Group Id as:
> The Provider Group Id is a unique interger ID that is defined by the user to be referenced in the [Negotiated Rate Details Object](https://github.com/CMSgov/price-transparency-guide/tree/master/schemas/in-network-rates#negotiated-rate-details-object) in the provider_references array.

I don't expect it to be unique across different issuers. But for a given issuer, at what level is it intended to be unique? I have seen cases where issuers have multiple in-network files referenced by a single table of contents file.  Also seen cases where a single issuer has multiple TOC file (each in turn, referencing multiple in-network files). and there are cases where an issuer (Aetna is an example) has broken up their files by line of business.

**So my questions are:**

* for @shaselton-usds: at what level does the spec intend the `provider_group_id` to be unique?
* for payers that are reading this:  when creating your files are you treating `provider_group_id` as unique across all of your data or only within some smaller namespace?

**Comment by shaselton-usds** [comment, marked as answer]
Hey @theogaraj, thanks for the question -

The in-network files should contain all of the necessary information to stand alone. This means that whether an in-network file leverages [external provider reference](https://github.com/CMSgov/price-transparency-guide/tree/master/schemas/provider-reference) files (which is recommended for both sizing issues and reusability) or the [`provider_references`](https://github.com/CMSgov/price-transparency-guide/tree/master/schemas/in-network-rates#in-network-file) array at the root of the in-network file, the `provider_group_id` should only apply to the data that is in that file. There isn't any other way for the consumers of the files to match a `provider_reference` object that would be defined outside of the in-network file.

**Comment by theogaraj** [comment]
@shaselton-usds thanks for the response.  

I was originally going to disagree with the rationale, but your reasoning makes sense now I realize that a given issuers files may include in-network files generated by another entity (could be leased or shared networks).  So Issuer A's TOC might reference files generated by Issuer A as well as files generated by Issuer B (where Issuer A has no control on what identifiers are used by Issuer B).
